### ISSUE TITLE
xmobar: error while loading shared libraries: libHSlibmpd-0.9.0.7-14aXvoHDkR6BCSO8ZXIrgk-ghc8.2.2.so: cannot open shared object file: No such file or directory

### ISSUE THREAD

**Issue by chrissound** [jaor/xmobar#329]
I've just upgraded my arch linux packages and I'm getting the following error on startup:
```
xmobar: error while loading shared libraries: libHSlibmpd-0.9.0.7-14aXvoHDkR6BCSO8ZXIrgk-ghc8.2.2.so: cannot open shared object file: No such file or directory
```

I've gotten similar issues like this before - probably something to do with Arch Linux and it's dynamic / static Haskell packages.

**Comment by 0xdefaced** [comment]
you could try to rebuild your package database cache `ghc-pkg recache`

**Comment by x-ji** [comment]
I have the AUR package `xmobar-git` installed and encountered this bug after a system update. Reinstalling the AUR package fixed it.

**Comment by chrissound** [comment]
Thanks for that, @x-ji. I also had the same package installed, seems to happen to a bunch of other packages too.

**Comment by chrissound** [comment]
@anthraxx Well that is useful to finally know, perhaps a better place to mention this would be the Haskell Arch wiki?

**Comment by x-ji** [comment]
@anthraxx This is the first thing that comes up when one searches the error message on the internet, so I thought it might just be better to directly point out the solution, so that other people searching the message know what to do. Now I know fully well why one will have to rebuild the tree, but such information is not given on the AUR git package page. I don't see how we're "reporting this here" by just pointing out one should deal with the issue, and not reopening the issue. Sorry if you're somehow offended.

**Comment by anthraxx** [comment]
It's actually not limited to git or haskell packages, but any AUR package... This combination (especially haskell based AUR packages) are only affected by soname changes more often, more precisely every time one of its dependencies changes.
Reality is, literally any package installed from AUR that links to something could be affected by the same problem you are facing here. Whenever a soname is bumped in an official repo manage library all custom packages linking against it need to be rebuilt.
This is what you buy when installing something  from AUR, it's unofficial, unsupported and you need to take care of such rebuilds yourself. People just tend to forget when they grab a random AUR helper and use the AUR in a same fashion as they use official packages.

**Comment by x-ji** [comment]
True, true. Were it not for the random crashes I would not have installed the AUR package either.

**Comment by OpAmlee** [comment]
same problem.. packages were out of sate. updated pacman and fixed. thank you -THE NOOB
